File: Algorithm/Programmers/Python/Level 1/시저 암호.py
def solution(s, n):
    answer = ''
    for i in s:
        if i.islower():
            answer+=push(i,'z',n)
        elif i.isupper():
            answer+=push(i,'Z',n)
        elif i==' ':
            answer+=' '
    
    return answer

def push(c,z,d):
    if ord(c)+d>ord(z):
        return chr(ord(c)+d-26)
    else:
        return chr(ord(c)+d)

File: Algorithm/Programmers/Java/Level 1/시저_암호.java
class 시저_암호 {
    public String solution(String s, int n) {

        char[] ch = s.toCharArray();

        for (int i = 0; i < s.length(); i++) {
            if (ch[i] >= (int) 'a' && ch[i] <= (int) 'z') {
                int tmp = (int) ch[i] + n;
                if (tmp > (int) 'z') {
                    tmp = tmp + 'a' - 'z' - 1;
                }
                ch[i] = (char) tmp;
            } else if (ch[i] >= (int) 'A' && ch[i] <= (int) 'Z') {
                int tmp = (int) ch[i] + n;
                if (tmp > (int) 'Z') {
                    tmp = tmp + 'A' - 'Z' - 1;
                }
                ch[i] = (char) tmp;
            }
        }

        return new String(ch);
    }
}